buildroot/tests/run_tests

```diff
@@ -14,7 +14,11 @@ exec_test () {
     printf "\033[0;32mPassed\033[0m\n"
     return 0
   else
-    git reset --hard HEAD
+    if [[ $GIT_RESET_HARD == "true" ]]; then
+      echo git reset --hard HEAD
```

Comment:
Oops!

Yeah, when I started running them I realised it would reset my changes, which when you're developing would be devastating.

Two questions on that front:
* Should we backup any files that we touch, so that in case you're developing new switches on `Configuration.h`/`_adv.h` you don't loose your changes?
* We do `chmod +x buildroot/bin/* buildroot/tests/*`. Would id be OK if I changed the permissions on all of those files, so that we don't have to change anything during tests?

Reply:
> Should we backup any files that we touch, so that in case you're developing new switches on Configuration.h/_adv.h you don't loose your changes?

I think it is ok to expect Configuration changes to be committed prior to testing. Some type of save/restore could be added later if it ends up being annoying.

> We do chmod +x buildroot/bin/* buildroot/tests/*. Would id be OK if I changed the permissions on all of those files, so that we don't have to change anything during tests?

I think changing them would be fine. I personally don't even know how to do that in git!
It might not work perfectly as you expect for users on Windows who then run tests inside WSL or MinGW.